### ISSUE TITLE
[NPC] Shirrak the Dead Watcher 

### ISSUE THREAD

**Issue by Rushor** [dalaranwow/dalaran-wow#3831]
[//]: # (INFORMATION ABOUT TAGS)
[//]: # (The black and yellow tag is unique. There can only be one label assigned to them - one black and one yellow)
[//]: # (OLD - ... tags should no longer be used)
[//]: # (Priority - ... tags are assigned by team members)

[//]: # (***************************************************************)
[//]: # (** !!!!!!!!!!!!!DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED!!!!!!!!!!!! **)
[//]: # (***************************************************************)

**Description**:
http://de.wowhead.com/npc=18371/shirrak-der-totenwachter#encounter
**Current behaviour**: Tell us what happens in detail (Where, When, What, Why, Who)
The spellscript for Inhibit Magic is not working properly
**Expected behaviour**: Tell us what should happen instead
http://de.wowhead.com/npc=18371/shirrak-der-totenwachter#encounter
Inhibit Magic: This aura places a debuff on all players in its range, which increases casting time by 50% per stack. Stacks up to 4 times for a total of +200% to cast time. The maximum level of stacking is related to the distance the player stands from the boss; the closer you stand the higher it stacks. It is possible to get the debuff before entering combat. Casters can (and should) stand at maximum range to minimize the effect of this debuff. Receiving this debuff will also remove stealth when applied.
<45 m: 1 Stack
<35 m: 2 Stacks
<25 m: 3 Stacks
<15 m: 4 Stacks

**Include proofs for this behaviour**
http://de.wowhead.com/npc=18371/shirrak-der-totenwachter#encounter
You can research this on several pages like http://wowhead.com or http://wowwiki.wikia.com or http://www.youtube.com/
Issues without proofs will be closed and tag On Hold will be added until the proof is provided.

**Include the ID for the game objects, npcs (creatures,pets,minions), spells, items, quests, instances, zones, achievements, skills**
http://de.wowhead.com/npc=18371/shirrak-der-totenwachter#encounter
You can research this on several pages like http://wotlk.openwow.com or http://wowhead.com/

**Comment by Rushor** [comment]
Scripts/Auchenai Crypts: Shirrak the Dead Watcher - fix stacks of "Inhibit Magic"

* commit: 84403e9b92f87c8445eb3c523a9924cecf892b51

The issue is fixed on the Public Test Realm (PTR). The changes/fixes will be applied with the NEXT changelog on the liverealm.
This means that the issue exists in game at the moment i wrote this message.
Please watch http://www.dalaran-wow.com/news/ to see when the changelog and the fix was applied.

**Comment by TeridanDW** [comment]
The stacking of  "Inhibit Magic" works as intended on ptr.
However there is a way to exploit this mechanic and prevent the boss of ever casting spell http://wotlk.openwow.com/spell=32264
Video from PTR of how to reprocude the issue:
https://www.youtube.com/watch?v=74jtIkJsmT0&feature=youtu.be

 Additional issues with this creature:
 - the trigger where spell http://wotlk.openwow.com/spell=32300 lands is visible/attackle/is attacking to players and the animation is wrong, should be like here:
https://www.youtube.com/watch?v=0G_YPDO7VfI

**Comment by Rushor** [comment]
Yes the movement issue was resolved with https://github.com/dalaranwow/dalaran-wow/issues/3827

DB/Creature: Mark Focus Fire Spell Target Dummy as trigger in order to make him invisible

* updates: https://github.com/dalaranwow/dalaran-wow/issues/3831
* commit: 500d62b60861150ed7f8025288845ed2cc059bab

Please open up a new issue for the wrong animation spell visual.

The issue has been fixed by a developer and will be tested on the Public Test Realm in the future (PTR). The changes/fixes will be applied with the NEXT changelog on the liverealm in the future.
This means that the issue exists in game at the moment i wrote this message.
**Only once the issue was marked with "Fixed - On Live Server", you can find the fix on your game server.**
Please watch http://www.dalaran-wow.com/news/ to see when the changelog and the fix was applied.

**Comment by TeridanDW** [comment]
Hello after few tests on PTR the results are:

 - Even after #3827 is fixed the:

> However there is a way to abuse this mechanic and prevent the boss of ever casting spell http://wotlk.openwow.com/spell=32264

can still be done the same way like it was shown above.

 - This:
>  - Mark Focus Fire Spell Target Dummy as trigger in order to make him invisible

Made not only the trigger invisible but also the spell on the ground resulting in range player been hit and pushed from the ground by invisible effect.
